Adequacy Patch - Local matching

flomnes opened the pull request.

hugo-antoine-rtei committed:
Adequacy Patch - Local Matching Rule 15/03 (#597)

* adq ini file with default value false

* typo

* created the option button(false/true) for adequacy patch
and updated the ref-guide

* typo

* ui selector

* deleted boolean cast & clang-format on new files

* clang format

* adq patch mode

* clang format

* added new header file for DENS

* changes to the filles

* syntax error

* bring back original all.h for adequacy

* clang format

* assigning dummy values to DENS

* minor name change

* default value 1

* typo

* column name change and typo

* change dens dummy value to zero

* Adequacy patch: 2 rounds run / boundary update / output (#10)

* created AdequacyFirstStep attribute

* reference fix

* save DENS value from 1st run

* add ::SIM_RenseignementProblemeHebdo before 2nd run

* save DENS in result and show in UI

* refacto minor

* solution for upperbound update (#11)

* draft solution for upperbound update

* refacto

* minor refacto

* clean

* clean

* DENS only for type 2 area

* review modif

Co-authored-by: rtei-pw <pengbo.wang@rte-international.com>
Co-authored-by: rtei-pw <92968374+rtei-pw@users.noreply.github.com>

* NTC constraints management (#8)

* set NTC constraints to 0 in a dummy if loop

* Small changes to the dummy loops

* insert adqPatchMode data in PROBLEME_HEBDO struct

* 0<->0 connections allowed

* implement specifications logic into NTC constraint

* shift SetNTCForAdequacyFirstStep to separate *.cpp

* clang format

* Hugo's comments implementation

* clean

* UI NTC selector for adequacy patch (#12)

* update reference guide

* add options to problem.hebdo

* ui options for ntc constraints for adequacy patch

* naming and comments improvements

* ref.guide description improvement

* plug UI into SetNTCForAdequacyFirstStep

* Rte comments 1. 2. 11. implementation (#20)

* name changes in domesticUnsuppliedEnergy.h

* 11. enumeration

* review modif 3. 5. 7, 8 (#16)

* 5. consider previously set bounds

* 7. remove not multiple

* 8. add todo arround SIM_

* 3. namespace

* 4. change Start&End to Origine&Extremity

* 3. bis fix

Co-authored-by: milos <milos.andjelkovic@redtork-solutions.com>

* 10. make adq struct

* 11. additional changes for comment 11 (#22)

* 10. make adqPatch pointer

* group adq parameters in a sub category

* bug fix

* small typo

* change NTC name and avoid using directives

* placed NTC function inside a  AdqPatch namespace

* rename Ntc11 & Ntc12 to be more expressive

* change End and Start in ntc function

* use map for SetNTCForAdequacyFirstStep function

* use map for SetNTCForAdqFirstStep function update!

* make type std::pair<AdqPatchMode, AdqPatchMode>

* clang format

* make setBounds(No)AdqPatch functions

* r2 review changes part 1 (#30)

* review: These two ifs can be merged into one.

* review: Create a new function called OPT_OptimisationHebdomadaireAdqPatch, call it here.

* review: domestic unsupplied energy (comment)

* adq-patch is only for studyMode == economy

* move behaviorMap to AdequacyPatchParameters struct

* attempt at making map creation clearer

* using just the Antares::Data namespace

* adq runtime data (#35)

* adq runtime data

* conflict

* RTE Comment implementation 3 (#34)

* typo Economy

* add comment to explain type 1 and type 2 for adq

* using namespace AdequacyPatch inside adq_patch.cpp

* if there is no area in the adq-patch  throw error

* divide to smaller functions SetNTCForAdqFirstStep

* clang

* check that adq patch is on in error handling

* bug fix

Co-authored-by: rtei-pw <pengbo.wang@rte-international.com>
Co-authored-by: milos <milos.andjelkovic@redtork-solutions.com>
Co-authored-by: rtei-pw <92968374+rtei-pw@users.noreply.github.com>
Co-authored-by: Milos <97689304+Milos-RTEi@users.noreply.github.com>

guilpier-code commented on src/libs/antares/study/area/list.cpp:
```diff
@@ -1040,6 +1069,44 @@ static bool AreaListLoadFromFolderSingleArea(Study& study,
+    {
+        buffer.clear() << study.folderInput << SEP << "areas" << SEP << area.id << SEP
+                       << "adequacy_patch.ini";
+        IniFile ini;
```
We could make a function/method with the following code block : 
```c++
void readAdqPatchMode(area&, buffer)
```

guilpier-code commented on src/libs/antares/study/parameters.h:
```diff
@@ -502,6 +505,14 @@ class Parameters final
+    //! Transmission capacities from physical areas outside adequacy patch (area type 1) to physical
+    //! areas inside adequacy patch (area type 2). NTC is set to null (if true) only in the first
+    //! step of adequacy patch local matching rule.
+    bool setToZero12LinksForAdequacyPatch;
```
"12" or "11" are not expressive for anyone who did not read adequacy patch documentation.
We could rename these variables like : **setToZeroNTCfromOutToIn_AdqPatch**, which is not much longer.

flomnes replied:
Fixed in "last remarks..."

guilpier-code commented on src/solver/optimisation/opt_gestion_des_bornes_cas_lineaire.cpp:
```diff
@@ -177,9 +178,10 @@ void OPT_InitialiserLesBornesDesVariablesDuProblemeLineaire(PROBLEME_HEBDO* Prob
-            Xmax[Var] = ValeursDeNTC->ValeurDeNTCOrigineVersExtremite[Interco];
-
-            Xmin[Var] = -(ValeursDeNTC->ValeurDeNTCExtremiteVersOrigine[Interco]);
+            if (ProblemeHebdo->adqPatch && ProblemeHebdo->adqPatch->AdequacyFirstStep)
```
As you could see, this source file (like many others in the modeler) is messy because functions are too big.
We should avoid to add more lines to functions.
Particularly here : instead of having a **if... else...**, we should have a call to **setNTCbounds(...)**,
and put the **if... else...** inside.
Please also see the remarks regarding file **src\solver\optimisation\adequacy_patch.cpp**, they complete what I mean.

guilpier-code commented on src/solver/optimisation/adequacy_patch.cpp:
```diff
@@ -0,0 +1,146 @@
+      ProblemeHebdo->adqPatch->LinkCapacityForAdqPatchFirstStepFromAreaOutsideToAreaInsideAdq,
+      ProblemeHebdo->adqPatch->LinkCapacityForAdqPatchFirstStepBetweenAreaOutsideAdq);
+
+    if (SetToZeroLinkNTCForAdequacyPatchFirstStep == setToZero)
```
Here a **switch** would be better than a **if... elsif...elsif...else**, especially because of the **enum** variable to be tested.

guilpier-code commented on src/solver/optimisation/adequacy_patch.cpp:
```diff
@@ -0,0 +1,146 @@
+    }
+    else
+    {
+        Xmax = ValeursDeNTC->ValeurDeNTCOrigineVersExtremite[Interco];
```
Here you have a code duplication with the content of **setBoundsNoAdqPatch(...)**.
So you should consider calling this latter function from **setBoundsAdqPatch(...)**. 
Or, even better if it's possible, you could set **Xmin** and **Xmax** from **no adq patch** case as the default values before the **switch** statement, so that you would not have to treat the **no adq patch**
case in the **switch**.

flomnes replied:
done

guilpier-code commented on src/libs/antares/study/fwd.h:
```diff
@@ -557,4 +557,45 @@ class Sets;
+enum AdequacyPatchMode
+{
+    //! Virtual area in adq patch
+    adqmVirtualArea = 0,
```
What does the **adqm** suffixing each item mean ?
If you put **AdequacyPatchMode::** before items when using it, you probably won't need  **adqm**

guilpier-code commented on src/solver/optimisation/opt_gestion_des_bornes_cas_lineaire.cpp:
```diff
@@ -401,9 +403,18 @@ void OPT_InitialiserLesBornesDesVariablesDuProblemeLineaire(PROBLEME_HEBDO* Prob
                 else
                     Xmax[Var] = 0.;
 
+                // adq patch: update ENS <= DENS in 2nd run
```
To me, the code format is unclear. I would put it like this : 
```c++
if (ProblemeHebdo->adqPatch && 
    ProblemeHebdo->adqPatch->AdequacyFirstStep == false &&
    ProblemeHebdo->adequacyPatchRuntimeData.areaMode[Pays] == AdequacyPatch::adqmPhysicalAreaInsideAdqPatch)
{
    Xmax[Var] = min(Xmax[Var], ProblemeHebdo->ResultatsHoraires[Pays]->ValeursHorairesDENS[PdtHebdo]);
}
```
You can even make use of **using namespace** upper in the file to shorten enum values references here.

guilpier-code commented on src/solver/simulation/economy.cpp:
```diff
@@ -146,7 +174,16 @@ bool Economy::year(Progression::Task& progression,
         try
         {
-            OPT_OptimisationHebdomadaire(pProblemesHebdo[numSpace], numSpace);
+            if (pProblemesHebdo[numSpace]->adqPatch)
```
The choice of which problem to run should not be made here, but more in a construction phase. 
For instance in the **simulationBegin** function (upper in this file), or even better, in the **Economy** class constructor (if possible). 

A possible way of doing things would be  : 
- Creating an interface class **weeklyProblemInterface** with a **run** method.
- Creating 2 classes implementing **weeklyProblemInterface** :  **weeklyProblem** and **weeklyProblemWithAdequacyPatch**, with their own run method.
- As the "classic" **run** method would be called from the **weeklyProblemWithAdequacyPatch** (to avoid code duplication), this classic run could be implemented in the interface class, and called **"as is"** from the **weeklyProblem**'s run method.

Anyway, here we would only have : 
```c++
theWeeklyProblem->run(/*some arguments ?*/)
```
and that's it

guilpier-code commented on src/solver/optimisation/opt_gestion_des_bornes_cas_lineaire.cpp:
```diff
@@ -401,9 +403,18 @@ void OPT_InitialiserLesBornesDesVariablesDuProblemeLineaire(PROBLEME_HEBDO* Prob
                 else
                     Xmax[Var] = 0.;
 
+                // adq patch: update ENS <= DENS in 2nd run
```
Furthermore, I guess you had to find a loop on areas (Pays) to insert bound setting on ENS.
Still to simplify the code here, you could extract this code into a separate function, even if you would have to extract the loop on areas within it.

flomnes replied:
See issue #716

guilpier-code commented on src/solver/simulation/sim_calcul_economique.cpp:
```diff
@@ -63,6 +63,21 @@ void SIM_InitialisationProblemeHebdo(Data::Study& study,
 
+    if (parameters.include.adequacyPatch)
+    {
+        problem.adqPatch = std::unique_ptr<AdequacyPatchParameters>(new AdequacyPatchParameters());
```
Use of **make_unique** ?

flomnes replied:
Only available in c++17

guilpier-code commented on src/solver/simulation/sim_structure_probleme_economique.h:
```diff
@@ -317,6 +321,28 @@ typedef struct
+    std::vector<AdequacyPatchMode> extremityAreaType; 
+    void initialize(Antares::Data::Study& study)
+    {
+        for (uint i = 0; i != study.areas.size(); ++i)
```
- on one hand, we have area**Mode**, on the other we have originArea**Type**. From my understanding, these vectors contain the same kind of values. How about : **areasAdqPatchModes** and **originAreasAdqPatchModes** ?
- As there is 2 loops making different things, we could have 2 private methods to make them... Or not

guilpier-code commented on src/solver/optimisation/adequacy_patch.cpp:
```diff
@@ -0,0 +1,146 @@
+{
+    LinkCapacityForAdequacyPatchFirstStep SetToZeroLinkNTCForAdequacyPatchFirstStep;
+
+    SetToZeroLinkNTCForAdequacyPatchFirstStep = SetNTCForAdequacyFirstStep(
```
For me : 
- Too many arguments in this function, and it makes difficult to understand what the call does. Arguments here should be developed inside the function.
- The name of the function starts with "Set", but actually it is more a "get".
- This name could be shorten because, at this stage, we know we're in the adq patch first step, and it's about links. So "Links" and "first step" can be removed from the function name. 

I would put it this way :   
```c++
ntcToZeroStatusForAdqPatch = getNTCtoZeroStatus(Interco, ProblemeHebdo)
```

flomnes replied:
See #717

guilpier-code commented on src/solver/simulation/sim_structure_probleme_economique.h:
```diff
@@ -469,6 +496,13 @@ typedef struct
+struct AdequacyPatchParameters
+{
+    bool AdequacyFirstStep;
+    bool LinkCapacityForAdqPatchFirstStepFromAreaOutsideToAreaInsideAdq;
```
Name a bit long and unclear for me.
Suggestion :  
- setToZeroNTCfromInToOut_AdqPatchStep1
- setToZeroNTCbetweenOutsideAreas_AdqPatchStep1

guilpier-code commented on src/solver/optimisation/adequacy_patch.cpp:
```diff
@@ -0,0 +1,146 @@
+    case adqmPhysicalAreaInsideAdqPatch:
+        returnNTC = setToZero;
+        break;
+    case adqmPhysicalAreaOutsideAdqPatch:
```
Based on the documentation that I have, for a link of type **Area Inside** => **Area Outside** (that is 2 -> 1), its NTC is always set to zero.  So here, I suppose we should have : 
 ```C++
switch (ExtremityNodeAdequacyPatchType)
{
    case adqmPhysicalAreaInsideAdqPatch:
        returnNTC = setToZero;
        break;
    case case adqmPhysicalAreaOutsideAdqPatch:
       returnNTC = setExtremityOrigineToZero;
       break;
    default:
        returnNTC = leaveLocalValues;
        break;
}
```

flomnes replied:
I agree with @guilpier-code's remark

guilpier-code commented on src/libs/antares/study/fwd.h:
```diff
@@ -557,4 +557,45 @@ class Sets;
+/*!
+** \brief Setting Link Capacity (NTC) for Adequacy patch first step
+*/
+enum LinkCapacityForAdequacyPatchFirstStep
```
Wouldn't **ntcSetToZeroStatus_AdqPatchStep1** be better (more clear) ?

guilpier-code commented on src/solver/optimisation/adequacy_patch.cpp:
```diff
@@ -0,0 +1,146 @@
+        returnNTC = leaveLocalValues;
+        break;
+    }
+    return returnNTC;
```
You don't need a local variable **returnNTC**,  you should return directly the result of functions or return  **leaveLocalValues**

guilpier-code commented on src/solver/optimisation/adequacy_patch.cpp:
```diff
@@ -0,0 +1,146 @@
+  AdequacyPatchMode ExtremityNodeAdequacyPatchType,
+  bool SetToZero12LinksForAdequacyPatch)
+{
+    LinkCapacityForAdequacyPatchFirstStep returnNTC;
```
As above, you don't need **returnNTC**

guilpier-code commented on src/solver/optimisation/adequacy_patch.cpp:
```diff
@@ -0,0 +1,146 @@
+  bool SetToZero12LinksForAdequacyPatch,
+  bool SetToZero11LinksForAdequacyPatch)
+{
+    LinkCapacityForAdequacyPatchFirstStep returnNTC;
```
As above, you don't need **returnNTC**

guilpier-code commented on src/ui/simulator/application/study.cpp:
```diff
@@ -90,6 +90,7 @@ Event<void(Data::AreaLink*)> OnStudyLinkDelete;
 
 Event<void()> OnStudySimulationSettingsChanged;
 Event<void()> OnStudyNodalOptimizationChanged;
+Event<void()> OnStudyAreaUseAdequacyPatchChanged;
```
The event **OnStudyAreaUseAdequacyPatchChanged** does not seem to be connected to a function / treatment / behavior. So, does it have any effect ?

guilpier-code commented on src/ui/simulator/windows/inspector/accumulator.hxx:
```diff
@@ -906,6 +909,28 @@ struct PAreaUnsuppliedEnergyCost
+    {
+        return area->adequacyPatchMode;
+    }
+    static wxString ConvertToString(const Type v)
```
Why not : 
```C++
static wxString ConvertToString(const Type v)
{
     return adequacyPatchMode[v];
}
```
You defined **adequacyPatchMode** array earlier in this file.
Note that having a value out of the **enum** range is very unlikely to happen, precisely because **enum** types do not accept values out of their range. So instruction : 
```C++
return wxEmptyString
```
is dead code

guilpier-code commented on src/ui/simulator/windows/inspector/property.update.cpp:
```diff
@@ -172,6 +172,29 @@ bool InspectorGrid::onPropertyChanging_A(wxPGProperty*,
+                s.toLower();
+                s.trim();
+
+                if (s == "virtual area" || s == "0")
```
- we could avoid **if...else if...else if**, replacing it with a switch statement
- Apparently, **s** cannot contain anything else than 1 character : "0", "1" or "2", I tested it.

So we could have : 
```C++
switch (s.to<int>())
{
    case 0:
        area->adequacyPatchMode = Data::AdequacyPatch::adqmVirtualArea;
        break;
     case 1:
         area->adequacyPatchMode = Data::AdequacyPatch::adqmPhysicalAreaOutsideAdqPatch;
         break;
      case 2:
          area->adequacyPatchMode = Data::AdequacyPatch::adqmPhysicalAreaInsideAdqPatch;
          break;
      default:
          break;
}
``` 
I tested it, it works fine.

guilpier-code commented on src/ui/simulator/windows/options/optimization/optimization.h:
```diff
@@ -115,6 +116,9 @@ class Optimization final : public wxDialog
 
     Component::Button* pBtnExportMPS;
+    Component::Button* pBtnAdequacyPatch;
+    Component::Button* pBtnAdequacyPatchNTC12;
```
"12" and "11" unclear. Prefer **pBtnNTCfromOutToInAdqPatch**

guilpier-code commented on src/ui/simulator/windows/options/optimization/optimization.cpp:
```diff
@@ -46,6 +46,21 @@ namespace Window
 {
 namespace Options
 {
+static void SubTitle(wxWindow* parent, wxSizer* sizer, const wxChar* text, bool margintop = true)
```
- **SubTitle** is not a very expressive name about what the function does. Maybe **addLabelAdqPatch** would be better.
- As this function is used once  with **margintop** as **true**, **margintop** is probably useless. Consider removing this argument. Inside the function, keep what's inside the **if** but remove the **if** test.

guilpier-code commented:
From what I see, setting the adequacy patch status of an area (virtual area, physical area inside or outside the adequacy patch domain domain) can only be done from the inspector.
Am I right ?
If this is the case, it does seem to be a "compliant" use of GUI, maybe we should consider this.

guilpier-code commented on src/solver/application.cpp:
```diff
@@ -84,6 +84,36 @@ void checkSimplexRangeHydroHeuristic(Antares::Data::SimplexOptimization optRange
+{
+    if (adqPatchOn)
+    {
+        bool containsAdqArea = false;
```
Not sure this variable is useful : the exception could be directly raised from inside the loop over areas

flomnes replied:
See #654

guilpier-code commented on src/libs/antares/study/area/list.cpp:
```diff
@@ -326,6 +331,30 @@ bool saveAreaOptimisationIniFile(const Area& area, const Clob& buffer)
+    switch (area.adequacyPatchMode)
+    {
+    case Data::AdequacyPatch::adqmVirtualArea:
+        value = 0;
```
I can see that when saving any area's **adequacy-patch.ini**, we have something like : 
```ini
[adequacy-patch]
adequacy-patch-mode = 2
```
The value **2** is not very expressive. We could have **inside adq patch domain** for instance

flomnes committed:
Make absence of areas/<area>/adequacy_patch.ini non-fatal (#655)

Milos-RTEi committed:
make readAdqPatchMode function

Milos-RTEi committed:
make ini file more expressive

flomnes committed:
Cosmetic changes to the adequacy patch (pt. 1 - local matching rule) (#654)

* Write 1 in the ts-numbers when series.width == 1 (#609)

* Write 1 in the ts-numbers when series.width == 1

Before that patch, random numbers were drawn, ignored in the
simulation, but still written in ts-numbers.

* Remove useless conversion to long

* Revert "Remove useless conversion to long"

This reverts commit c43ab0b07741dff4e94d00bdc9dca5c9a58aff53.

* Revert "Write 1 in the ts-numbers when series.width == 1"

This reverts commit 8d32e2a3e08c27f93fea0df6c01beae0b034e469.

* Remove useless .entry

use operator[] instead

* Set timeseriesNumbers to 0 if width==1

* [skip ci] Spacing

* Remove failed tests

When intramodal is set to true, it can't be expected that all
time-series numbers be equal, even if width==1 for some clusters.

* [sonar] Fix sonar code smells

* PR remarks

* Revert "PR remarks"

This reverts commit b4abb71ef92a91d16476d8b329e387d14358618f.

* Store ts-numbers at the end of the simulation

* [skip ci] Add explanatory comment

* PR remarks

* [skip ci] Use std::for_each, code formatting

* Release 8.2.1 : changelog & version number

* Add noise to the cost in the daily & monthly allocation problem (#622)

* Antares solver options : fix a bug on Windows (#624)

* Antares solver options : fix a bug on Windows

* Antares solver options : correction due to sonarQ analysis

* Hydro levels initialization : relaxing error message in case reservoir management is OFF (#625)

* Use std::unique_ptr for GetOpt::Parser (#628)

Co-authored-by: Florian Omnès <florian.omnes@rte-france.com>

* Tests : activating end-to-end tests on Windows OS (#525)

* Tests : activating end-to-end tests on Windows OS

* Making tests run output more readable

* Allow pytest to display logs from print(...)

* Test too see if we display details on errors

* Refactoring code for comparison of simulation to reference, at least to remove this horrible recursive compare function.

* Comparison report improvements (see details after)

- Removing report's embarrassing error logs from a study execution
- Reducing report's number of lines per file to be printed down to 10
- Changing the report columns name (self ==> reference, other ==> results)
- Adding a column to the report : the differences between the 2 first columns

* Run tests on CI : fixing 'diff' column values (relevant values are the opposite)

* Run tests on CI : make possible to set tolerances for each test

* Run tests on CI : set tolerances for 2 tests

* Run tests on CI : allow sending std error to devnull for unfeasible problem tests

* Run tests on CI : corrections due to code review

* Run tests on CI : add an enlarged tolerance to test "playlist-psp-misc-ndg"

* Remove references to BUILD_not_system (#637)

This option has been moved to antares-deps. It no longer exists in Antares_Simulator.

* Scenario builder warnings too long (#626)

* Antares solver options : fix a bug on Windows

* Antares solver options : correction due to sonarQ analysis

* Hydro levels initialization : relaxing error message in case reservoir management is OFF

* sc builder warnings too long

Warnings from scenario builder can be very long :
this happens when a cluster (thermal or renewable) is disabled and when some TS numbers are assigned to this cluster in the scenario builder.
As the cluster is disabled, Antares warns the user of all years where a TS number is set.
When this year collection is grows, the warning line grows accordingly.

* sc builder warnings too long : correction due to review

Co-authored-by: Florian OMNES <florian.omnes@rte-france.com>

* Remove build option BUILD_OUTPUT_TEST (#636)

* Do not perform quadratic opt if FLOW QUAD. is disabled (#617)

in the thematic trimming.

* Fix GCC warnings (#554)

* constraintsbuilder.cpp : remove unused arguments

* Remove useless asserts &variable != NULL

* Fix compilation warnings in studylogs.{h/cpp}

* Fix compilation warnings in exportmap.cpp

* Remove unused asserts & function parameters

* Fix several compilation warnings

* control.{h,cpp} : Use in-class initializer

* Pass const char* instead of char*

* Remove unused variables

* Fix more compilation warnings

* Fix more compilation warnings

* Fix CMake

* Restore STUDY_MODE_EXPANSION

* Fix more compilation warnings

* Make constants 'const'

* Add 'explicit' keyword

* Revert change in yuni job queue

* PR remarks

* PR remarks

* Fix warnings in src/tests/libs/antares

* Fix possibily unitialized variable

* Remove unused 'warn' argument from Notebook::find

* Use std::size_t in tests

* Remove warning in debug mode

It wasn't displayed beforehand.

* Remove unused variable MapPage

* Remove commented-out code

* Fix : Wind & Solar time-series set as generated, but renewable policy  is "Clusters" (#650)

* Fix : Wind & Solar time-series set as generated, but renewable policy is "Clusters"

In case we're in the situation described by the commit title, a crash occurs at execution.
Program tries to copy a wind TS though it was not loaded nor generated.
So we avoid that copy (that we don't need anyway) when the renewable policy is not aggregated.

* Apply clang-format

Co-authored-by: Florian OMNES <florian.omnes@rte-france.com>

* [solver] Remove useless parentheses (#652)

find -type f | xargs sed -E -i 's/\((.+)\)->/\1->/g'

* Use in-class initialization

* Use std::any_of

* Use 'using namespace XXX' in switch..case

* Apply clang-format

* Remove useless event 'OnStudyAreaUseAdequacyPatchChanged'

This event does not trigger anything.

* Remove trailing whitespace

* Revert "Merge develop into fix/adq-patch-rte-small-changes"

This reverts commit 441d613ccc7f6c5582c2ce9dfba306711f17e75b, reversing
changes made to f93dc320701bcb49b1b7b31316301cd213a91a8b.

* Remove "using namespace" directives from header files

Co-authored-by: guilpier-code <62292552+guilpier-code@users.noreply.github.com>

flomnes committed:
Merge branch develop feature/adequacy-patch

flomnes committed:
Simplify switch..case

flomnes committed:
Add missing header file inclusions

flomnes committed:
Use switch..case instead of multiple if..else, use setBoundsNoAdqPatch

flomnes committed:
Force values inside enum

flomnes committed:
{origin=2} -> {extremity=1} return setToZero

flomnes committed:
Simplify switch..case

flomnes committed:
Rename adequacy patch member data

flomnes committed:
Remove useless parentheses

flomnes committed:
Fix build

flomnes reviewed (commented):
- [small] 11, 12 : not clear. Make sure to use 11 -> OutsideOutside, 12 -> OutsideInside
- [small] UI : Subtitle function : remove last argument whose value is always `true`
- [small] UI : switch case + no conversion to `YString` to parse enum from menu
- [medium] Interface class for the weekly problem (factory method ?), at least add explanatory comments

flomnes committed:
Merge branch develop

flomnes committed:
Apply clang-format

flomnes committed:
Add interface class + implementations for economy weekly problem (#712)

* Add interface class + implementations for economy weekly problem

* Add missing &

To avoid unwanted copies

* Revert "Add missing &"

This reverts commit bd5fd3ceb346f0b0917680293bd383cd9eb1f9cf.

* using EcoWeeklyPtr = std::unique_ptr<EconomyWeeklyOptimization>

* Move factory method into static method of the iface class

* Use EconomyWeeklyOptimization::Ptr

flomnes committed:
Add unit testing for adq-patch (local matching) (#699)

* Add unit testing for adq-patch (local matching)

* Avoid re-compiling adequacy_patch.cpp, link against existing target

* Fix build path

* Refactor tests to avoid code duplication

* Fix STL headers

* Add explanatory comments

* [CMake] Include tests only if recent Boost is found

* Making this branch compile under MSVC 2019

* Rename PROBLEME_HEBDO::adqPatch -> adqPatchParams

* Rename one_timestep -> setNTCboundsForOneTimeStep

* Rename adqmVirtualArea -> virtualArea, etc.

Co-authored-by: Guillaume PIERRE <guillaume.pierre@rte-france.com>

flomnes committed:
Merge develop into feature/adequacy-patch

flomnes committed:
Local matching rule : last remarks (#704)

* Add unit testing for adq-patch (local matching)

* Avoid re-compiling adequacy_patch.cpp, link against existing target

* Fix build path

* Refactor tests to avoid code duplication

* Fix STL headers

* Add explanatory comments

* [CMake] Include tests only if recent Boost is found

* Rename button variables

* pBtnAdequacyPatchNTC12 -> pBtnAdqPatchOutsideInside
* pBtnAdequacyPatchNTC11 -> pBtnAdqPatchOutsideOutside

* Rename 11 -> OutsideInside, 12 -> OutsideInside

* Apply clang-format

* Remove useless margintop arg

* Do not try to convert value to string, convert to long

guilpier-code commented on src/solver/simulation/sim_structure_probleme_economique.h:
```diff
@@ -469,6 +497,13 @@ typedef struct
     double* CoutsMarginauxHorairesDeLaReserveParZone;
 } COUTS_MARGINAUX_ZONES_DE_RESERVE;
 
+struct AdequacyPatchParameters
```
ok

guilpier-code reviewed (commented):
ok
